code/test/tokens.js

```diff
@@ -0,0 +1,472 @@
+    await testerc20.transfer(alice.wallet.address, 1000, { from: alice.lastResortAddress })
+    aliceWalletBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    assert.equal(1000, aliceWalletBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet succesful')
+    // alice transfers tokens to bob
+    await tokenTransfer(
+      {
+        wallet: alice,
+        operationType: ONEConstants.OperationType.TRANSFER_TOKEN,
+        tokenType: ONEConstants.TokenType.ERC20,
+        contractAddress: testerc20.address,
+        dest: bob.wallet.address,
+        amount: 100
+      }
+    )
+    // OneWallet Items that have changed
+    // check alice and bobs balance
+    aliceBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    aliceWalletBalanceERC20 = await alice.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    bobBalanceERC20 = await testerc20.balanceOf(bob.wallet.address)
+    bobWalletBalanceERC20 = await bob.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    assert.equal(900, aliceBalanceERC20, 'Transfer of 100 ERC20 tokens from alice.wallet succesful')
+    assert.equal(900, aliceWalletBalanceERC20, 'Transfer of 100 ERC20 tokens from alice.wallet succesful and wallet balance updated')
+    assert.equal(100, bobBalanceERC20, 'Transfer of 100 ERC20 tokens to bob.wallet succesful')
+    assert.equal(100, bobWalletBalanceERC20, 'Transfer of 100 ERC20 tokens to bob.wallet succesful and wallet balance updated')
+    // Alice Items that have changed - nonce, lastOperationTime, Commits, trackedTokens
+    assert.notStrictEqual(await alice.wallet.nonce, alice.state.nonce, 'alice wallet.nonce should have been changed')
+    // alice.state.nonce = parseInt(alice.state.nonce) + 1
+    alice.state.nonce = await alice.wallet.getNonce()
+    // lastOperationTime
+    assert.notStrictEqual(await alice.wallet.lastOperationTime(), alice.state.lastOperationTime, 'alice wallet.lastOperationTime should have been updated')
+    alice.state.lastOperationTime = await alice.wallet.lastOperationTime()
+    // commits
+    assert.notDeepEqual(await alice.wallet.getAllCommits(), alice.state.allCommits, 'alice wallet.allCommits should have been updated')
+    alice.state.allCommits = await alice.wallet.getAllCommits()
+    // tracked tokens
+    assert.notDeepEqual(await alice.wallet.getTrackedTokens(), alice.state.trackedTokens, 'alice.wallet.trackedTokens should have been updated')
+    alice.state.trackedTokens = await alice.wallet.getTrackedTokens()
+    await CheckUtil.checkONEWallet(alice.wallet, alice.state)
+    // Bob Items that have changed - nothing
+    await CheckUtil.checkONEWallet(bob.wallet, bob.state)
+  })
+
+  // ERC20 Decimals 9 Testing (Transfer, Mint, Track, SpendingLimit)
+  it('Wallet_CommitReveal: ERC20-9(Transfer, Mint, Track) must commit and reveal successfully', async () => {
```

Comment:
The purpose of Decimals-9 test contract is primarily to test whether tokens with less decimals (e.g. USDT) are displayed correctly in the UI (in terms of amount). The focus would be on the conversion functions (ONEUtil.toFraction, etc.). So a separate test at the smart contract level is probably not needed, unless it involves testing formatting and value conversion functions

Reply:
Agreed - removed